src/commands/Changelog.ts

```diff
@@ -0,0 +1,318 @@
+                        );
+                    } else {
+                        throw new Error(
+                            "Programming mistake: No existing expression mapping for a syntax node that is not new: " +
+                                entry.expression
+                        );
+                    }
+                }
+            }
+
+            writeStream.write(
+                EOL +
+                    "Already known and still used syntax nodes which were removed from or added to some language(s):" +
+                    EOL +
+                    EOL
+            );
+            writeStream.write(
+                "Name:" +
+                    csvSeparator +
+                    "Added to language(s):" +
+                    csvSeparator +
+                    "Removed from language(s):" +
+                    csvSeparator +
+                    "Remains in language(s):" +
+                    csvSeparator +
+                    "Used for calculating the metric(s):" +
+                    csvSeparator +
+                    "Was explicitly only activated for language(s):" +
+                    EOL
+            );
+
+            for (const entry of this.getAll().values()) {
+                if (
+                    entry.remainingLanguages.size > 0 ||
+                    (entry.addedLanguages.size > 0 && entry.removedLanguages.size > 0)
+                ) {
+                    const mapping = metricMappings.get(entry.expression);
+                    if (mapping !== undefined) {
+                        const activatedForLangOutput =
+                            mapping.activated_for_languages === undefined
+                                ? ""
+                                : mapping.activated_for_languages;
+                        writeStream.write(
+                            escapeForCsv(entry.expression) +
+                                csvSeparator +
+                                Array.from(entry.addedLanguages) +
+                                csvSeparator +
+                                Array.from(entry.removedLanguages) +
+                                csvSeparator +
+                                Array.from(entry.remainingLanguages) +
+                                csvSeparator +
+                                mapping.metrics +
+                                csvSeparator +
+                                activatedForLangOutput +
+                                EOL
+                        );
+                    } else {
+                        throw new Error(
+                            "Programming mistake: No existing expression mapping for a syntax node that is not new: " +
+                                entry.expression
+                        );
+                    }
+                }
+            }
+            writeStream.end();
+        });
+    }
+}
+
+/**
+ * Tracks the changes that were applied to one expression.
+ */
+class ChangelogEntry {
```

Comment:
Since this class is not exported, the name should be fine. Otherwise I would rename it to express that this is more like an ExpressionChangelogItem or so.

Reply:
Renamed to NodeTypesChangelog(Entry), just like the name of the file it produces